scripts/functions.py

```diff
@@ -106,4 +107,404 @@ def daylightSavings(inData):
         newRow = [rowAfter[0], rowAfter[1], rowAfter[2], newHour, rowAfter[4]]
         inData.insert(i, newRow)
 
-    return inData
\ No newline at end of file
+    return inData
+
+def initializeSeasons():
+    # PURPOSE: Initializes seasons as a dictionary
+    # INPUT: None
+    # OUTPUT: seasons (dictionary)
+
+    # Dictionary holds month to season Mapping
+    seasons = {
+        'W':[1, 2, 12],
+        'SP':[3, 4, 5],
+        'S':[6, 7, 8],
+        'F':[9, 10, 11]}
+
+    return seasons
+
+def initializeYears():
+    # PURPOSE: Initializes years as a list
+    # INPUT: None
+    # OUTPUT: years (list)
+
+    #Years to Print over
+    dfYears = pd.read_csv('../src/data/Canada/YEAR.csv')
+    years = dfYears['VALUE'].tolist()
+
+    return years
+
+def initializeRegions():
+    # PURPOSE: Initializes regions as a list
+    # INPUT: None
+    # OUTPUT: regions (list)
+
+    # Regions to print over
+    dfRegions = pd.read_csv('../src/data/Canada/REGION.csv')
+    regions = dfRegions['VALUE'].tolist()
+    return regions
+
+def initializeSubregions():
+    # PURPOSE: Initializes subregions as a dictionary
+    # INPUT: None
+    # OUTPUT: subregions (dictionary)
+
+    #Dictionary for subregion to province mappings
+    subregions = defaultdict(list)
+
+    # Read in regionalization file to get provincial seperation
+    df = pd.read_excel('../dataSources/Regionalization.xlsx', sheet_name='CAN')
+    for i in range(len(df)):    
+        subregion = df['REGION'].iloc[i]
+        province = df['PROVINCE'].iloc[i]
+        subregions[subregion].append(province)
+    
+    return subregions
+
+def initializeSubregionsWithNoDuplicates():

```

Comment:
Can probably just call initializeSubregions? Initialization assumes it removes duplicates?

Reply:
Also, might be good to specify this is only canada subregions?